api/src/main/java/io/cloudevents/extensions/ExtensionsParser.java

```diff
@@ -0,0 +1,41 @@
+package io.cloudevents.extensions;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.Extension;
+
+import java.util.HashMap;
+import java.util.function.Supplier;
+
+public final class ExtensionsParser {
+
+    private static class SingletonContainer {
+        private final static ExtensionsParser INSTANCE = new ExtensionsParser();
+    }
+
+    public static ExtensionsParser getInstance() {
+        return SingletonContainer.INSTANCE;
+    }
+
+    private HashMap<Class<?>, Supplier<Extension>> extensionFactories;
```

Comment:
class -> impl is known to be hard to work with.

Have you considered a set of factories with `supports(Class<T> extensionClass)` method?